scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/task/TaskStateImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.ow2.proactive.scheduler.task;
+
+import java.util.List;
+
+import org.ow2.proactive.scheduler.common.task.TaskInfo;
+import org.ow2.proactive.scheduler.common.task.TaskState;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+
+public class TaskStateImpl extends TaskState {
+
+    private TaskInfo taskInfo;
+    private int maxNumberOfExecutionOnFailure = PASchedulerProperties.NUMBER_OF_EXECUTION_ON_FAILURE.getValueAsInt();
+    private int iterationIndex;
+    private int replicationIndex;
+    
+    @Override
+    public void update(TaskInfo taskInfo) {
+        this.taskInfo = taskInfo;
+    }
+
+    @Override
+    public List<TaskState> getDependences() {
+        return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)